lib/reqTemplate.js

```diff
@@ -0,0 +1,143 @@
+"use strict";
+
+var URI = require('swagger-router').URI;
+var rbUtil = require('./rbUtil');
+var TAssembly = require('tassembly');
+
+/**
+ * Creates a function that sets a value at the path and initializes all objects/arrays along the path.
+ *
+ * @param path the path within an object/array separated with dots {e.g. test.body.a.b.c}
+ * @returns {Function} the function to set a value at path and initialize object along the path if needed
+ * @private
+ */
+function _setAtPath(path, spec) {
+    var pathArr = path.split('.');
+    var refArr = [];
+    var subspec = spec;
+    var code;
+
+    for (var idx = 0; idx < pathArr.length; idx++) {
+        var curRef = {};
+        curRef.ref = Array.isArray(subspec) ? '[' + pathArr[idx] + ']' : '["' + pathArr[idx] + '"]';
+        subspec = subspec[pathArr[idx]];
+        curRef.initializer = Array.isArray(subspec) ? '[]' : '{}';
+        refArr.push(curRef);
+    }
+
+    code =  'if (value !== undefined) {\n';
+    for (var i = 1; i < refArr.length; i++) {
+        var ref = refArr.slice(0, i).map(function(elem) { return elem.ref; }).join('');
+        code += 'obj' + ref + ' = obj' + ref + ' || ' + refArr[i - 1].initializer + ';\n';
+    }
+    code += 'obj' + refArr.map(function(elem) { return elem.ref; }).join('') + ' = value;\n';
+    code += '}';
+
+    /* jslint evil: true */
+    return new Function('obj', 'value', code);
+}
+
+/**
+ *  Constructs a list of resolvers for all templates in a request spec
+ *
+ * @param subspec a request spec or subspec for a part of request
+ * @param reqPart name of request part (e.g. headers, body, query)
+ * @param path path to the current subspec within a full spec
+ * @returns {Array} an array of template resolvers for every template found in the spec.
+ *          Resolvers are functions with (newRequest, request) arguments that modify newRequest param.
+ * @private
+ */
+function _createTemplateResolvers(origSpec, subspec, reqPart, path) {
+
+    /**
+     * Makes an array of resolvers for every template found within a template spec object.
+     *
+     * @param templateSpec A subspec of an original request spec.
+     * @param templatePath A path to the current subspec
+     * @returns {Array} and array of resolvers for each template found in the subspec
+     */
+    function makeResolvers(templateSpec, templatePath) {
+        var setValue = _setAtPath(templatePath, origSpec);
+        if (templateSpec instanceof Object) {
+            return _createTemplateResolvers(origSpec, templateSpec, reqPart, templatePath);
+        } else if (/^\{[^}]+}$/.test(templateSpec)) {
+            var template = /^\{([^}]+)}$/.exec(templateSpec)[1];
+            var buildModel;
+
+            if (!/^(\$\.)?([a-zA-Z0-9-_]*)(\.[a-zA-Z0-9-_]*)*$/.test(template)) {
```

Comment:
I think we should require each component to start with a letter, and not a number.